docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,102 @@
+```
+                 +------------+
+                 |            |
+                 | Management |
+                 | API        |
+                 |            |
+                 |            |
+                 ++--+------+-+
+                  ^  ^      ^
+                  |  |      |
+         +--------+  |      +--------------+
+         |           |                     |
+         v           v                     v
++--------+-+      +--+-------+           +-+--------+
+|          |      |          |           |          |
+| Delivery |      | Delivery |           | Delivery |
+| Channel  |      | Channel  | +-+  +-+  | Channel  |
+| 1        |      | 2        |           | N        |
+|          |      |          |           |          |
+|          |      |          |           |          |
++----------+      +----------+           +----------+
+```
+
+The management api should have knowledge of what delivery channels are available and where certain notification types should be sent. 
+
+When a new notification for a user is sent to the management api, the management api will read the user preference for the said notification type and see which communication types the user has 'subscribed' to. Following this, the notification delivered through the appropriate delivery channel.
+
+### User notifications and user settings
+
+A user should be able to choose what kind of communication types they receive based on the notification type. There are two ways of approaching this, either:
+1. Send different types of notifications through all channels that are provided by the consumer of this service, and leave the communication type and user configuration for the consumer of the Marain.UserNotifications service. (ie, the consumer of the Marain.UserNotifications service should specify which delivery channels they want to send notifications via an api call when a notification is created)
+2. Store user preferences somewhere that is accessible to the management api, then when a new notification with a certain notification type comes through, check what type of communication types  the user has 'subscribed' to and then send a message through the delivery channel.
+
+The latter option was chosen as it provides a user the ability to choose what notifications they want out of the box, if the former is chosen, all users/businesses integrating with this service will then need to write their own notifications user management service.
+
+
+
+Settings for which communication channel a user can receive a notification through will be stored inside a User Notifications Settings table which will be accessed by the Management Api. Configuration in this table will include the Notification Type and the appropriate communication types associated to that notification type. This table will have an associated crud api that will allow a consumer of `Marain.UserNotifications` to view/change the stored settings.
+
+### Management Api and usable Delivery Channels
+
```

Comment:
I'd re-order this slightly.

First, I'd describe how delivery channels will be implemented, and how the management API will be able to enumerate them (this is missing at the moment, I think - I left a separate comment about it, but it wasn't attached to anything so got lost).

Then describe how a tenant can "enable" or "configure" a delivery channel for a communication type.

Then describe how the user preferences that control where and how specific notification types are maintained.

Reply:
I've re-ordered the document slightly to fit your above comment. Should I also add a header that will then describe exactly how to add configuration (api keys and associated communication types) for a delivery channel for a certain tenant.